pkg/proxy/handlers/spacelister_get_test.go

```diff
@@ -248,27 +246,29 @@ func testSpaceListerGet(t *testing.T, publicViewerEnabled bool) {
 								AvailableActions: []string(nil), // this is system generated so no actions for the user
 							},
 						}),
-					),
+					)}
 				},
 				expectedErr:       "",
 				expectedWorkspace: "movielover",
 			},
 			"get nstemplatetier error": {
 				username:        "dance.lover",
-				expectedWs:      []toolchainv1alpha1.Workspace{},
+				expectedWs:      nil,
 				expectedErr:     "nstemplatetier error",
 				expectedErrCode: 500,
-				overrideInformerFunc: func() service.InformerService {
-					informerFunc := fake.GetInformerService(fakeClient, fake.WithGetNSTemplateTierFunc(func(_ string) (*toolchainv1alpha1.NSTemplateTier, error) {
-						return nil, fmt.Errorf("nstemplatetier error")
-					}))
-					return informerFunc()
+				mockFakeClient: func(fakeClient *test.FakeClient) {
+					fakeClient.MockGet = func(ctx context.Context, key runtimeclient.ObjectKey, obj runtimeclient.Object, opts ...runtimeclient.GetOption) error {
+						if key.Name == "base1ns" {
```

Comment:
minor - alternatively we could check (also) the object type.

Same for the other similar checks.

Reply:
We cannot use `TypeMeta` as this is empty. we would need to either check the actual go type of the object, or get the kind from the schema. TBH, both of these options seem to be an overkill for these unit tests. I don't think that we would return an error in other cases than the expected ones.
but I'm fine with adding it there if you see some possible problems with the current approach.

Reply:
casting and checking the type in this way is actually simple and can be done in single-line. so I updated the mocks: 0194c4c

Reply:
right, that is what I was thinking. Thanks for addressing my comment.